src/System.Net.Security/src/System/Net/_SecureChannel.cs

```diff
@@ -0,0 +1,1487 @@
+                        else
+                            Logging.PrintInfo(Logging.Web, this, SR.net_log_remote_cert_has_no_errors);
+                    }
+                    else
+                    {
+                        if (remoteCertValidationCallback != null)
+                            Logging.PrintInfo(Logging.Web, this, SR.net_log_remote_cert_user_declared_invalid);
+                    }
+                }
+                GlobalLog.Print("Cert Validation, remote cert = " + (remoteCertificateEx == null ? "<null>" : remoteCertificateEx.ToString(true)));
+            }
+            finally
+            {
+                // At least on Win2k server the chain is found to have dependencies on the original cert context.
+                // So it should be closed first.
+                if (chain != null)
+                {
+                    chain.Dispose();
+                }
+                if (remoteCertificateEx != null)
+                    remoteCertificateEx.Dispose();
+            }
+            GlobalLog.Leave("SecureChannel#" + Logging.HashString(this) + "::VerifyRemoteCertificate", success.ToString());
+            return success;
+        }
+
+        internal static uint Verify(SafeX509ChainHandle chainContext, ref Interop.Crypt32.CERT_CHAIN_POLICY_PARA cpp)
+        {
+            GlobalLog.Enter("SecureChannel::VerifyChainPolicy", "chainContext=" + chainContext + ", options=" + String.Format("0x{0:x}", cpp.dwFlags));
+            var status = new Interop.Crypt32.CERT_CHAIN_POLICY_STATUS();
+            status.cbSize = (uint)Marshal.SizeOf<Interop.Crypt32.CERT_CHAIN_POLICY_STATUS>();
+
+            bool errorCode =
+                Interop.Crypt32.CertVerifyCertificateChainPolicy(
+                    (IntPtr)Interop.Crypt32.CertChainPolicy.CERT_CHAIN_POLICY_SSL,
+                    chainContext,
+                    ref cpp,
+                    ref status);
+
+            GlobalLog.Print("SecureChannel::VerifyChainPolicy() CertVerifyCertificateChainPolicy returned: " + errorCode);
+#if TRAVE
+            GlobalLog.Print("SecureChannel::VerifyChainPolicy() error code: " + status.dwError + String.Format(" [0x{0:x8}", status.dwError) + " " + Interop.MapSecurityStatus(status.dwError) + "]");
+#endif
+            GlobalLog.Leave("SecureChannel::VerifyChainPolicy", status.dwError.ToString());
+            return status.dwError;
+        }
+
+        public Interop.Secur32.SecureCredential CreateSecureCredential(
+            int version,
+            X509Certificate certificate,
+            Interop.Secur32.SecureCredential.Flags flags,
+            int protocols, EncryptionPolicy policy)
+        {
+            var credential = new Interop.Secur32.SecureCredential()
+            {
+                rootStore = IntPtr.Zero,
+                phMappers = IntPtr.Zero,
+                palgSupportedAlgs = IntPtr.Zero,
+         
```

Comment:
GitHub places your comment on line 
`+                certContextArray = IntPtr.Zero,`

This is a ctor and so I presume you're referring to the `GlobalLog.Leave` line? There's a TRACE_VERBOSE statement above that.